### PR TITLE
feat: annotate codebase for storage config options

### PR DESCRIPTION
**Pull request by gabrielmougard**
Fix the [Storage drivers](https://documentation.ubuntu.com/lxd/en/latest/reference/storage_drivers/)  configuration to be generated from the code. This means, for all storage drivers:

* Move the descriptions to the code
* Replace the table in the docs with an include from the generated options file
* Search and replace any links to those properties with a config reference

JIRA ticket: https://warthogs.atlassian.net/browse/LXD-441

blocked by: https://github.com/canonical/lxd/pull/12642

### CODE REVIEW

**Comment by tomponline** [comment]
@gabrielmougard for the benefit of people who dont have access to jira (and for future reference) please can you expand the PR description to explain what this PR does? Thanks

File: doc/config_options.txt [hunk 1]
```diff
@@ -1832,3 +1832,1157 @@ This value is required by some providers.
...
+
+```
+
+```{config:option} security.shifted storage-btrfs-storage-volume-conf
+:condition: "custom volume"
+:defaultdesc: "same as `volume.security.shifted` or `false`"
+:shortdesc: "Enable ID shifting overlay (allows attach by multiple isolated instances)"
+:type: "bool"
+
+```
```
Comment (ru-fu) [inline doc/config_options.txt:1897]: The same option is included three times!

File: doc/config_options.txt [hunk 2]
```diff
@@ -1832,3 +1832,1157 @@ This value is required by some providers.
...
+:defaultdesc: "same as `volume.lvm.stripes`"
+:shortdesc: "Number of stripes to use for new volumes (or thin pool volume)"
+:type: "string"
+
+```
+
+```{config:option} lvm.stripes.size storage-ceph-storage-volume-conf
+:defaultdesc: "same as `volume.lvm.stripes.size`"
+:shortdesc: "Size of stripes to use (at least 4096 bytes and multiple of 512 bytes)"
+:type: "string"
```
Comment (ru-fu) [inline doc/config_options.txt:2109]: Probably related to the other issue, but these are not ceph-related.

**Comment by ru-fu** [comment]
I have now added some commits to the branch to fix the config option descriptions and include them in the documentation. Please check and squash as required. :)

There's an issue with duplicate entries that must be resolved before we can merge.

**Review by ru-fu — CHANGES_REQUESTED** [review]
Need to fix the duplicate entries.

**Comment by ru-fu** [comment]
Uhm, you just force-pushed over all the commits I added. Let me get them back.

**Comment by ru-fu** [comment]
Updated again. Please squash as required, but don't remove the changes. ;)

**Comment by gabrielmougard** [comment]
> Uhm, you just force-pushed over all the commits I added. Let me get them back.

sorry for that.

**Comment by gabrielmougard** [comment]
@ru-fu updated with your changes

File: doc/config_options.txt [hunk 3]
```diff
@@ -1832,3 +1832,848 @@ This value is required by some providers.
...
+
+```
+
+```{config:option} lvm.stripes.size storage-ceph-volume-conf
+:defaultdesc: "same as `volume.lvm.stripes.size`"
+:shortdesc: "Size of stripes to use"
+:type: "string"
+The size must be at least 4096 bytes, and a multiple of 512 bytes.
+```
```
Comment (ru-fu) [inline doc/config_options.txt:2031]: Did you regenerate the file? These options should not be there for Ceph.

**Review by ru-fu — CHANGES_REQUESTED** [review]

File: doc/config_options.txt (hunk 3 above)
Comment (gabrielmougard) [reply doc/config_options.txt:2031]: Ah yes good catch! I'm fixing it

Comment (gabrielmougard) [reply doc/config_options.txt]: @ru-fu should be ok now

**Review by ru-fu — DISMISSED** [review]
Thanks!

**Review by tomponline — APPROVED** [review]

**Comment by tomponline** [comment]
@gabrielmougard needs a rebase please

**Comment by tomponline** [comment]
needs a rebase again please

**Comment by tomponline** [comment]
@ru-fu please can you let me know if happy with this and ill merge.

**Comment by tomponline** [comment]
@edlerd this wont break the existing UI right?

**Comment by edlerd** [comment]
> @edlerd this wont break the existing UI right?

Nope, it shall not break anything. Quite the contrary, this unblocks our task to use the config api for the storage form elements.

**Comment by tomponline** [comment]
Thanks

@gabrielmougard please can this be rebased now the earlier PR is merged. Thanks

**Review by ru-fu — APPROVED** [review]

**Review by tomponline — APPROVED** [review]

**Pull request merged**